Ensures the correct format of the phone number as required by the Stancer API

nola-adrien opened the pull request:
Ensures the correct format of the phone number as required by the Stancer API.

For instances:
FR number "01 12 45 87 98" becomes "+33112458798"
DE number "(0) 641/460 99-0" becomes "+49641460990"

nola-adrien committed:
Ensures the correct format of the phone number as required by the Stancer API.
For instances
FR number "01 12 45 87 98" becomes "+33112458798"
DE number "(0) 641/460 99-0" becomes "+49641460990"

sderen-stancer commented on Gateway/Http/Client/AbstractTransaction.php:
Why did you format the phone number in abstract transaction, it think it would be more suited to do that in the Request components, as it does not mutate data, but only format them (like the amount field in `PaymentDataBuilder`)

sderen-stancer commented on Helper/PhoneFormatter.php:
```diff
@@ -0,0 +1,66 @@
+	 *
+	 * @return array
+	 */
+	public function updatePhoneNumbers(array $data): array
```
To add on my first comment, if the PhoneFormatter is called on the request layer we could use it directly on the necessary field, and we wouldn't need the method `updatePhoneNumbers`, but only `updatePhoneNumber`.

sderen-stancer commented on Gateway/Http/Client/AbstractTransaction.php:
```diff
@@ -38,6 +39,11 @@ abstract class AbstractTransaction implements ClientInterface
     protected $adapter;
 
+    /**
+     * @var
```
```suggestion
     * @var PhoneFormatter
```

sderen-stancer commented on Helper/PhoneFormatter.php:
```diff
@@ -0,0 +1,66 @@
+	}
+
+	/**
+	 * @param string $phoneNumber
```
```suggestion
	 * Format a phone number following E164 standard.
	 * @param string $phoneNumber A phone number with an unknown format.
```

sderen-stancer commented on Helper/PhoneFormatter.php:
```diff
@@ -0,0 +1,66 @@
+use Magento\Sales\Model\Order;
+
+/**
+ * Checkout workflow helper
```
If the class name is self explanatory you can leave the doc block empty.

sderen-stancer commented on Helper/PhoneFormatter.php:
```diff
@@ -0,0 +1,66 @@
+	 * @param string $phoneNumber
+	 * @param string $countryCode
+	 *
+	 * @return string
```
```suggestion
	 * @return string A phone number formatted following E164 standard.
```

sderen-stancer reviewed (changes requested):
Thank you for proposing an improvement for our magento module. Here is some suggestion and question about your merge request.

nola-adrien committed:
Ensures the correct format of the phone number as required by the Stancer API.
For instances
FR number "01 12 45 87 98" becomes "+33112458798"
DE number "(0) 641/460 99-0" becomes "+49641460990"
(feedbacks)

nola-adrien commented:
Thanks for your feedbacks @sderen-stancer , it makes sense.
I reverted the changes in AbstractTransaction, simplified PhoneFormatter and now update the phone in Gateway/Request/CustomerDataBuilder.php

sderen-stancer reviewed (approved):
Thank you, this is good for us.
